default.nix

```diff
@@ -1,14 +1,40 @@
 {
   niv ? import nix/sources.nix,
-  pkgs ? import niv.nixpkgs { config = { allowUnfree = true; }; }
+  nixpkgs ? niv.nixpkgs,
+  pkgs ? import nixpkgs {}
 }:
+
 let
-  cbsLib = import ./lib { inherit pkgs; };
-  cbspkgs = import ./pkgs { inherit niv pkgs cbsLib; };
-  cbsModules = import ./modules { inherit niv cbspkgs pkgs; };
-in cbspkgs //
-  {
-    lib = cbsLib;
-    nixpkgs = pkgs;
-    modules = cbsModules;
-  }
+  lib = import ./lib { inherit pkgs; };
+  initrdSet = import ./pkgs/initrd-creator/release.nix {
+    inherit pkgs;
+    cbsLib = lib;
+  };
+  kernelSet = import ./pkgs/kernels { inherit pkgs; };
+in pkgs.recurseIntoAttrs rec {
```

Comment:
Can you explain why you use `rec` here ? 

I do not see any recursive use of an attribute  from this set.

Reply:
Good catch. Fixed!